src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/MeshResponse.java

```diff
@@ -0,0 +1,23 @@
+package uk.nhs.digital.nhsconnect.nhais.mesh;
+
+import java.io.IOException;
+
+import lombok.RequiredArgsConstructor;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.util.EntityUtils;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@RequiredArgsConstructor
+class MeshResponse {
+    private final CloseableHttpResponse response;
+
+    public <T> T parseInto(Class<T> clazz) throws IOException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonParser parser = objectMapper.reader().createParser(EntityUtils.toString(response.getEntity()));
```

Comment:
This might be tricky. You enclose deserialization into a class passing in response as constructor param. The problem that might occur here is that you can read response only. Might be better to not to do it this way. Other thing it's a potential place for memory leak

Reply:
OK, I'll move it into private method